src/AzureAppConfigurationImpl.ts

```diff
@@ -168,11 +175,44 @@ export class AzureAppConfigurationImpl extends Map<string, any> implements Azure
      */
     async load() {
         await this.#loadSelectedAndWatchedKeyValues();
-
         // Mark all settings have loaded at startup.
         this.#isInitialLoadCompleted = true;
     }
 
+    /**
+     * Construct hierarchical data object from map.
+     */
+    constructConfigurationObject(options?: ConfigurationObjectConstructionOptions): Record<string, any> {
+        const separator = options?.separator ?? ".";
+
+        // construct hierarchical data object from map
+        const data: Record<string, any> = {};
+        for (const [key, value] of this.#configMap) {
+            const segments = key.split(separator);
+            let current = data;
+            // construct hierarchical data object along the path
+            for (let i = 0; i < segments.length - 1; i++) {
+                const segment = segments[i];
+                // undefined or empty string
+                if (!segment) {
+                    throw new Error(`invalid key: ${key}`);
+                }
+                // create path if not exist
+                if (current[segment] === undefined) {
+                    current[segment] = {};
+                }
+                // The path has been occupied by a non-object value, causing ambiguity.
+                if (typeof current[segment] !== "object") {
+                    throw new Error(`The key '${segments.slice(0, i + 1).join(separator)}' is not a valid path.`);
+                }
+                current = current[segment];
+            }
+            // set value to the last segment
+            current[segments[segments.length - 1]] = value;

```

Comment:
What will happen if there is a key "a.  ”?
data[a][" "] this can work. But for this method, I think our expected usage will be data.a.xxx

Reply:
Make sense.